### ISSUE TITLE
Bug: type-only imports cause parser error

### ISSUE THREAD

**Issue by dan-wellman-cs** [typescript-eslint/typescript-eslint#10420]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

i can't use the playground as it doesn't allow .vue files 

### Repro Code
I have a minimal config (in the new flat style), only slightly different from the config you recommend at https://typescript-eslint.io/getting-started/:

```TypeScript
import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';
import pluginVue from 'eslint-plugin-vue';

export default tseslint.config(
	eslint.configs.recommended,
  	tseslint.configs.recommended,
	...pluginVue.configs['flat/recommended'],
);
```

The only difference is the inclusion of the recommended Vue config, as I'm working with a Vue3 app.

But, I am not able to use any type imports in script blocks inside .vue SFC files my app. 
This causes a parser error:

```
import type { PropType } from 'vue';
```

If I remove the `type` part, then I instead get a warning that the import is a type-only import and I should import it using a type-only import.

### ESLint Config

```javascript
import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';
import pluginVue from 'eslint-plugin-vue';

export default tseslint.config(
	eslint.configs.recommended,
  	tseslint.configs.recommended,
	...pluginVue.configs['flat/recommended'],
);
```

### tsconfig

I'm not using type-aware linting

### Expected Result

I expect that type-only imports should not cause a parser error when used inside .vue files

### Actual Result

Type-only imports do cause a parser error when used inside .vue files

### Additional Info

Type-only imports seem like the kind of thing that should "just work" if everything is configured correctly, which it seems to be, because I can lint js/ts/vue files which don't have type-only imports.
